AutoCollection/HttpRequests.ts

```diff
@@ -82,28 +84,39 @@ class AutoCollectHttpRequests {
                 throw new Error('onRequest handler must be a function');
             }
             return (request:http.ServerRequest, response:http.ServerResponse) => {
+                const shouldCollect: boolean = !(<any>request)[AutoCollectHttpRequests.alreadyAutoCollectedFlag];
```

Comment:
I wonder if it's possible for `request` to be undefined here. Looks like we guard for it later